### PR TITLE
added support for amphp

### PR DESCRIPTION
**Pull request by umbri**
added support for https://github.com/amphp using https://github.com/amphp/artax

### CODE REVIEW

**Comment by unreal4u** [comment]
Oh this is great, it shows that the concept of decoupling the request handler with the telegram code actually did achieve its goal!

I'll merge this, but I'll have to revert commit a300731 for now, mainly due to its backward compatibility problem, and because I'm thinking of creating a dedicated namespace for multiple request handlers, so I would like to take both changes into v4, but I'll have to give that last one a bit more thought. I don't want to rush v4 too much given that v3 has been out in the open for less than a week .

Thanks!

**Comment by umbri** [comment]
@unreal4u dedicated namespace for multiple request handlers will be perfect, when it's done I will push with all needed changes to support [amphp](https://github.com/amphp/amp)

File: examples/00.send-message-amp.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+\Amp\Loop::run(function () {
+    $logger = (new \Monolog\Logger("log"))->pushHandler(new StreamHandler(STDOUT, Logger::ERROR));
+    $tgLog = new TgLog(BOT_TOKEN, new \unreal4u\TelegramAPI\HttpClientRequestHandlerAmp(), $logger);
+
+    $sendMessage = new SendMessage();
+    $sendMessage->chat_id = A_USER_CHAT_ID;
+    $sendMessage->text = 'Hello world!';
+
+    $tgLog->performApiRequest($sendMessage);
```
Comment (kelunik) [inline examples/00.send-message-amp.php]: This line should probably be `yield`ed, so errors bubble up.

File: src/HttpClientRequestHandlerAmp.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        $this->client = new DefaultClient;
+    }
+
+    /**
+     * @param string $uri
+     *
+     * @return PromiseInterface
+     */
+    public function get(string $uri): PromiseInterface {
+         $request = (new Request($uri))->withMethod('GET');
```
Comment (kelunik) [inline src/HttpClientRequestHandlerAmp.php]: You don't need `->withMethod`, `GET` is used by default. For other methods, you can just use the second constructor argument.

**Comment by umbri** [comment]
@kelunik done, I will PR

**Pull request merged**